Description: Don't check for cargo-vendor when building from (Debian's) git
Author: Matthijs van Otterdijk <matthijs@wirevirt.net>
Forwarded: not-needed
---
--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -915,7 +915,10 @@
         }
 
         // If we're building from git sources, we need to vendor a complete distribution.
-        if builder.rust_info.is_git() {
+        //
+        // Debian: disabling this block because the debian package is also in a git
+        //         repository, but cargo-vendor should not be installed or run.
+        if false && builder.rust_info.is_git() {
             // Vendor all Cargo dependencies
             let mut cmd = Command::new(&builder.initial_cargo);
             cmd.arg("vendor")
--- a/src/bootstrap/channel.rs
+++ b/src/bootstrap/channel.rs
@@ -30,7 +30,11 @@
 impl GitInfo {
     pub fn new(ignore_git: bool, dir: &Path) -> GitInfo {
         // See if this even begins to look like a git dir
-        if !dir.join(".git").exists() {
+        //
+        // Debian: force-enabling this block because the debian package is also in a git
+        //         repository, but we don't want to parse gitinfo. This is needed for the
+        //         bootstrap tests to work which running for Debian git.
+        if true && !dir.join(".git").exists() {
             return GitInfo::Absent;
         }
 
